src/roswire/common/package.py

```diff
@@ -116,116 +69,19 @@ class PackageDatabase(ABC, Mapping[str, Package]):
         and `db['foo'] = bar`).
     """
 
-    @classmethod
-    def build(
-        cls, app_instance: "AppInstance", paths: Optional[List[str]] = None
-    ) -> "PackageDatabase":
-        if paths is None:
-            paths = cls._determine_paths(app_instance)
-        db_package = cls._from_paths(app_instance, paths)
-        return db_package
+    _paths_in_package: List[str]
 
     @classmethod
     @abstractmethod
-    def _determine_paths(cls, app_instance: "AppInstance") -> List[str]:
-        """Parses the package paths for a given shell.
-
-        Parameters
-        ----------
-        app_instance: AppInstance
-            An instance of the application for which the
-            list of paths should be obtained
-        """
+    def build(cls,
+              app_instance: "AppInstance",
+              paths: Optional[List[str]]) -> "PackageDatabase":
```

Comment:
Nitpick: return type should be on a separate line.
More importantly, `paths` should have a default value of `None`.

Reply:
Done